torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1572,6 +1572,14 @@ def _lazy_init(self) -> None:
             # ``optim.step()`` is done before we all-gather parameters.
             self._wait_for_previous_optim_step()
 
+            # Set a flag on every FlatParameter to track whether its post
+            # backward hook has been called, mainly for validation purpose
+            # in wait_for_post_backward.
+            if self._exec_order_data.is_first_iter:
```

Comment:
Note to myself: I will think about where to move this logic with the future refactors.

Reply:
Thank you! We can probably adjust this further when lazy init cleanup lands: https://github.com/pytorch/pytorch/pull/80185